crates/astria-composer/tests/blackbox/composer.rs

```diff
@@ -161,7 +207,83 @@ async fn invalid_nonce_failure_causes_tx_resubmission_under_different_nonce() {
 }
 
 #[tokio::test]
-async fn single_rollup_tx_payload_integrity() {
+async fn invalid_nonce_failure_causes_tx_resubmission_under_different_nonce_grpc_collector() {
+    use crate::helper::mock_sequencer::mount_abci_query_mock;
+
+    // Spawn a composer with a mock sequencer and a mock rollup node
+    // Initial nonce is 0
+    let rollup_id = RollupId::from_unhashed_bytes("test1");
+    let test_composer = spawn_composer(&[rollup_id.to_string().as_str()]).await;
+    tokio::time::timeout(
+        Duration::from_millis(100),
+        test_composer.setup_guard.wait_until_satisfied(),
+    )
+    .await
+    .expect("setup guard failed");
+
+    // Reject the first transaction for invalid nonce
+    let invalid_nonce_guard =
+        mount_broadcast_tx_sync_invalid_nonce_mock(&test_composer.sequencer, rollup_id).await;
+
+    // Mount a response of 0 to a nonce query
+    let nonce_refetch_guard = mount_abci_query_mock(
+        &test_composer.sequencer,
+        "accounts/nonce",
+        NonceResponse {
+            height: 0,
+            nonce: 1,
+        },
+    )
+    .await;
+
+    let expected_chain_ids = vec![rollup_id];
+    // Expect nonce 1 again so that the resubmitted tx is accepted
+    let valid_nonce_guard =
+        mount_broadcast_tx_sync_mock(&test_composer.sequencer, expected_chain_ids, vec![1]).await;
+
+    // Send a tx to the composer so that it is picked up by the generic collector and submitted with
+    // the stored nonce of 0, triggering the nonce refetch process
+    let tx = Transaction::default();
+    // send sequence action request to the grpc generic collector
+    let mut composer_client = GrpcCollectorServiceClient::connect(format!(
+        "http://{}",
+        test_composer.grpc_collector_addr
+    ))
+    .await
+    .unwrap();
+    composer_client
+        .submit_rollup_transaction(SubmitRollupTransactionRequest {
+            rollup_id: rollup_id.as_ref().to_vec(),
+            data: tx.rlp().to_vec(),
+        })
+        .await
+        .expect("error submitting sequence actions to generic collector");
+
+    // wait for 1 sequencer block time to make sure the bundle is preempted
+    tokio::time::timeout(
+        Duration::from_millis(test_composer.cfg.block_time_ms),
+        invalid_nonce_guard.wait_until_satisfied(),
+    )
+    .await
+    .expect("invalid nonce guard failed");
+
+    tokio::time::timeout(
+        Duration::from_millis(100),
+        nonce_refetch_guard.wait_until_satisfied(),
+    )
+    .await
+    .expect("nonce refetch guard failed");
+
+    tokio::time::timeout(
+        Duration::from_millis(100),
+        valid_nonce_guard.wait_until_satisfied(),
+    )
+    .await
+    .expect("valid nonce guard failed");
```

Comment:
all these `expect` messages don't state why you expected these tests to work

Reply:
i ll update all the expect messages

Reply:
updating them to make more sense